test/integration/integration_test.go

```diff
@@ -0,0 +1,90 @@
+// +build integration
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testutil "sigs.k8s.io/cluster-api-provider-aws/test/commons"
+	"sigs.k8s.io/cluster-api/util"
+)
+
+var _ = Describe("integration tests", func() {
+	var (
+		namespace   string
+		clusterName string
+	)
+
+	BeforeEach(func() {
+		namespace = "test-namespace-" + util.RandomString(6)
+		testutil.CreateNamespace(namespace, kindClient)
+	})
+
+	Describe("cluster name validation tests", func() {
```

Comment:
It looks like we are testing something that should be part of a validation webhook. The way the validation webhook is structured we should be able to write a unit test for this logic without need an entire CAPA deployment.

Reply:
Do you think it's worthwhile to have automated testing for webhook validation someplace though?

Reply:
I thought we didn't have validation webhooks for release-0.4

Reply:
We don't, ha, I wasn't paying attention to the branch.

Reply:
If we write the actual logic for testing this stuff and it's complex then yes, a test would be nice. A unit test would be ideal for a function that is doing validation. Spinning up an entire kubernetes cluster and CAPA to see if a validation function passes is a lot and we should minimize overhead where possible.

If we don't have webhooks this logic will have to exist ... I guess in the aws cluster reconciler?

If we can use kubebuilder annotations for it that would also be fine with me. We wouldn't have to write tests for those, we can assume those work.

Reply:
I don't think we need a test to verify the name is a DNS-1123 subdomain. This is handled by kube-apiserver code.